File: apps/desktop/src/api/templates.ts
import { callCommand } from "./tauri";

export type SkillTemplateKind = "basic" | "playbook";

export type SkillTemplate = {
  kind: SkillTemplateKind;
  name: string;
  description: string;
  tags: string[];
  skillMd: string;
};

export function listSkillTemplates(): Promise<SkillTemplate[]> {
  return callCommand<SkillTemplate[]>("list_skill_templates");
}

export function getSkillTemplate(kind: SkillTemplateKind): Promise<SkillTemplate> {
  return callCommand<SkillTemplate>("get_skill_template", { kind });
}

File: apps/desktop/src-tauri/src/commands/templates.rs
use flowmint_core::store::template_store::{
    SkillTemplate, SkillTemplateKind, get_skill_template as core_get_skill_template,
    list_skill_templates as core_list_skill_templates,
};

#[tauri::command]
pub fn list_skill_templates() -> Vec<SkillTemplate> {
    core_list_skill_templates()
}

#[tauri::command]
pub fn get_skill_template(kind: SkillTemplateKind) -> SkillTemplate {
    core_get_skill_template(kind)
}
